### PR TITLE
[Commoncrawl pipeline] Add load from commoncrawl component

### PR DESCRIPTION
**Pull request by shayorshay**
This component is the first part of the commoncrawl pipeline. Given an index, this component loads the corresponding index file from the AWS Public Data Sets (S3 bucket) and returns a list of its WARC segment file paths as a dataframe.

### CODE REVIEW

File: examples/pipelines/commoncrawl/components/load_from_commoncrawl/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+boto3==1.26.158
+fondant
+pyarrow>=7.0
```
Comment (PhilippeMoussalli) [inline examples/pipelines/commoncrawl/components/load_from_commoncrawl/requirements.txt:3]: I think pyarrow is installed with Fondant so no need to include it here

File: examples/pipelines/commoncrawl/components/load_from_commoncrawl/src/main.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import gzip
+
+import dask.dataframe as dd
+import pandas as pd
+
+from fondant.component import LoadComponent
+
+logger = logging.getLogger(__name__)
+
+S3_BASE_URL = "s3://commoncrawl/crawl-data"
```
Comment (PhilippeMoussalli) [inline examples/pipelines/commoncrawl/components/load_from_commoncrawl/src/main.py:16]: This is not being used anywhere

File: examples/pipelines/commoncrawl/components/load_from_commoncrawl/src/main.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+S3_COMMONCRAWL_BUCKET = "commoncrawl"
+
+
+def fetch_warc_file_from_s3(s3_bucket: str, s3_key) -> dd.DataFrame:
```
Comment (PhilippeMoussalli) [inline examples/pipelines/commoncrawl/components/load_from_commoncrawl/src/main.py:20]: Can you add a docstring for the argumens. 

Would also prefer to change `s3_key` to `bucket_path`. Feels more intuitive

**Review by PhilippeMoussalli — APPROVED** [review]
Thanks @shayorshay ! Looks like a very clean component ;) 
Left a few minor comments but should be good to go

File: examples/pipelines/commoncrawl/components/load_from_commoncrawl/src/main.py (hunk 3 above)
Comment (shayorshay) [reply examples/pipelines/commoncrawl/components/load_from_commoncrawl/src/main.py:20]: `s3_key` is the key of the object in the bucket not the bucket path. Would `object_key` be a better name?

Comment (PhilippeMoussalli) [reply examples/pipelines/commoncrawl/components/load_from_commoncrawl/src/main.py:20]: my bad, just read on it and it seems to be an AWS specific notation (still used to GCP). It's fine to leave it as is

Comment (shayorshay) [reply examples/pipelines/commoncrawl/components/load_from_commoncrawl/src/main.py:20]: ok great!

File: examples/pipelines/commoncrawl/components/load_from_commoncrawl/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+boto3==1.26.158
+fondant
```
Comment (RobbeSneyders) [inline examples/pipelines/commoncrawl/components/load_from_commoncrawl/requirements.txt:2]:
```suggestion
git+https://github.com/ml6team/fondant@main
```

**Review by RobbeSneyders — APPROVED** [review]
Thanks @shayorshay, one minor comment since our latest version of the build script expects fondant dependencies to be defined from github.

File: examples/pipelines/commoncrawl/components/load_from_commoncrawl/requirements.txt (hunk 4 above)
Comment (shayorshay) [reply examples/pipelines/commoncrawl/components/load_from_commoncrawl/requirements.txt]: done :)

**Pull request merged**